src/awkward/contents/unionarray.py

```diff
@@ -1574,7 +1574,8 @@ def _remove_structure(
         self, backend: Backend, options: RemoveStructureOptions
     ) -> list[Content]:
         out = []
-        for i in range(len(self._contents)):
+        _, unique_index, *_ = numpy.unique_all(self._tags.data)
```

Comment:
@pfackeldey - should it be Numpy-like? To cover the `cupy` as a backend?

Reply:
yes, this is to cover all backends (also typetracer).

Reply:
oh, wait, you're right. I'm using the global `numpy` nplike. Let me have a look again to use the nplike of the content class... 🤦